src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1168,36 +1214,52 @@ int fmi3_xml_handle_IntXXVariable(fmi3_xml_parser_context_t* context, const char
             type = (fmi3_xml_int_type_props_t*)declaredType;
         }
         variable->type = &type->super;
+        /* TODO handle this buffer clean-up in a better way */
+        {
+            const char* tmp; /* unused */
+            fmi3_xml_get_attr_str(context, fmi3_xml_elmID_Int64, fmi_attr_id_start, 0, &tmp);
+        }
+    }
+    else {
+        /* set start value */
 
-        hasStart = fmi3_xml_get_has_start(context, variable);
+        /* We must wait until after parsing dimensions, because we can't otherwise know
+         * if it's a 1x1 array or a scalar variable just by reading the start value. */
+        int hasStart = variable->startAttr != NULL;
         if (hasStart) {
-            fmi3_xml_variable_start_int_t * start = (fmi3_xml_variable_start_int_t*)fmi3_xml_alloc_variable_type_start(td, &type->super, sizeof(fmi3_xml_variable_start_int_t));
-            fmi3_int_union_t defVal;
+            jm_string startAttr = variable->startAttr;
+
+            fmi3_xml_variable_start_int_t* start = \
+                    (fmi3_xml_variable_start_int_t*)fmi3_xml_alloc_variable_type_start(
+                        td,
+                        variable->type,
+                        sizeof(fmi3_xml_variable_start_int_t));
             if (!start) {
-                fmi3_xml_parse_fatal(context, "Could not allocate memory");
+                fmi3_xml_parse_fatal(context, "Could not allocate memory for an integer start array");
```

Comment:
I don't think we should add specific message to memory allocation failure. It doesn't really matter where it happens if it happens, and it's easier to just mentally ignore this kind of code if it's the same everywhere.

Reply:
I agree but I liked the idea that you dont have to attach a debugger in order to find where the error happens (when it happens). Do you prefer that I remove the recently added piece?

Reply:
Yes, for consistency. And my point is, if you get out-of-memory, it typically won't be deterministic where it happens, so the details of where it happens don't really matter.